### PR TITLE
#19455 Simplify conflate signature for the common case

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

**Comment by drewhk** [comment]
missed doc samples...

**Comment by drewhk** [comment]
failure is unrelated: https://github.com/akka/akka/issues/19585

**Comment by johanandren** [comment]
PLS BUILD

**Comment by johanandren** [comment]
Needs rebase

**Comment by drewhk** [comment]
Ok, I will do it

**Comment by drewhk** [comment]
now rebased.

**Comment by drewhk** [comment]
and now needs review

**Comment by drewhk** [comment]
refs #19445

**Comment by johanandren** [comment]
LGTM!

File: akka-docs/rst/scala/code/docs/stream/cookbook/RecipeSimpleDrop.scala [hunk 1]
```diff
@@ -15,11 +15,11 @@ class RecipeSimpleDrop extends RecipeSpec {
 
       //#simple-drop
       val droppyStream: Flow[Message, Message, NotUsed] =
-        Flow[Message].conflate(seed = identity)((lastMessage, newMessage) => newMessage)
+        Flow[Message].conflateWithSeed(seed = identity)((lastMessage, newMessage) => newMessage)
       //#simple-drop
       val latch = TestLatch(2)
       val realDroppyStream =
-        Flow[Message].conflate(seed = identity)((lastMessage, newMessage) => { latch.countDown(); newMessage })
+        Flow[Message].conflateWithSeed(seed = identity)((lastMessage, newMessage) => { latch.countDown(); newMessage })
```
Comment (2m) [inline akka-docs/rst/scala/code/docs/stream/cookbook/RecipeSimpleDrop.scala]: [Docs page](http://doc.akka.io/docs/akka/2.4.2-RC2/scala/stream/stream-cookbook.html#Dropping_elements) which includes this sample code mentions `conflate` a couple of times. It should now reference `conflateWithSeed` or the code should keep `conflate` and remove the `seed=identity` part.

**Comment by 2m** [comment]
Needs a sync with the cookbook. Otherwise LGTM.

File: akka-docs/rst/scala/code/docs/stream/cookbook/RecipeSimpleDrop.scala (hunk 1 above)
Comment (johanandren) [inline akka-docs/rst/scala/code/docs/stream/cookbook/RecipeSimpleDrop.scala]: The stages-overview should also mention it, but I can add that in the stages overview update ticket.

**Pull request merged**